onnxruntime/contrib_ops/cuda/bert/skip_layer_norm.cc

```diff
@@ -106,25 +106,29 @@ Status SkipLayerNorm<T, Simplified>::ComputeInternal(OpKernelContext* ctx) const
     }
   }
 
-  int sequence_length = static_cast<int>(input_dims[1]);
-  int hidden_size = static_cast<int>(input_dims[2]);
-  int64_t element_count = input_dims[0] * sequence_length * hidden_size;
-  size_t element_size = sizeof(T);
-  typedef typename ToCudaType<T>::MappedType CudaT;
+  int sequence_length = gsl::narrow_cast<int>(input_dims[1]);
+  int hidden_size = gsl::narrow_cast<int>(input_dims[2]);
+  int row_count = gsl::narrow_cast<int>(input_dims[0] * sequence_length);
 
-  return LaunchSkipLayerNormKernel<CudaT, Simplified>(
+  typedef typename ToCudaType<T>::MappedType CudaT;
+  HostApplyLayerNorm<CudaT, float, CudaT, Simplified>(
```

Comment:
Maybe we should just limit this change to GPT2-style SLN with a TODO to explore moving over altogether from current SLN implementation. Not sure what the accuracy and latency implications are for other models that have historically been using the "original" SLN implementation (for example BERT) ?

Reply:
Hope future optimization should on the Single LayerNorml Kernel.